unity/PingPongGameExample/PingPongGame/Assets/Scripts/Integration/PlatformIntegration/CarrierInfoIntegration.cs

```diff
@@ -0,0 +1,362 @@
+      }
+      return mccmnc;
+    }
+
+    KeyValuePair<string, uint> GetCidKeyValuePair(AndroidJavaObject cellInfo)
+    {
+      KeyValuePair<string, uint> pair = new KeyValuePair<string, uint>(null, 0);
+
+      AndroidJavaObject cellInfoClass = cellInfo.Call<AndroidJavaObject>("getClass", emptyObjectArr);
+      string simpleName = cellInfoClass.Call<string>("getSimpleName", emptyObjectArr);
+
+      if (simpleName.Equals(cellInfoTdscdmaString))
+      {
+        AndroidJavaObject cellIdentity = cellInfo.Call<AndroidJavaObject>("getCellIdentity", emptyObjectArr);
+        int cid = cellIdentity.Call<int>("getCid", emptyObjectArr);
+        pair = new KeyValuePair<string, uint>(simpleName, (uint)cid);
+      }
+      else if (simpleName.Equals(cellInfoNrString))
+      {
+        AndroidJavaObject cellIdentity = cellInfo.Call<AndroidJavaObject>("getCellIdentity", emptyObjectArr);
+        int nci = cellIdentity.Call<int>("getNci", emptyObjectArr);
+        pair = new KeyValuePair<string, uint>(simpleName, (uint)nci);
+      }
+      else if (simpleName.Equals(cellInfoLteString))
+      {
+        AndroidJavaObject cellIdentity = cellInfo.Call<AndroidJavaObject>("getCellIdentity", emptyObjectArr);
+        int ci = cellIdentity.Call<int>("getCi", emptyObjectArr);
+        pair = new KeyValuePair<string, uint>(simpleName, (uint)ci);
+      }
+      else if (simpleName.Equals(cellInfoGsmString))
+      {
+        AndroidJavaObject cellIdentity = cellInfo.Call<AndroidJavaObject>("getCellIdentity", emptyObjectArr);
+        int cid = cellIdentity.Call<int>("getCid", emptyObjectArr);
+        pair = new KeyValuePair<string, uint>(simpleName, (uint)cid);
+      }
+      else if (simpleName.Equals(cellInfoWcdmaString))
+      {
+        AndroidJavaObject cellIdentity = cellInfo.Call<AndroidJavaObject>("getCellIdentity", emptyObjectArr);
+        int cid = cellIdentity.Call<int>("getCid", emptyObjectArr);
+        pair = new KeyValuePair<string, uint>(simpleName, (uint)cid);
+      }
+      else if (simpleName.Equals(cellInfoCdmaString))
+      {
+        AndroidJavaObject cellIdentity = cellInfo.Call<AndroidJavaObject>("getCellIdentity", emptyObjectArr);
+        int baseStationId = cellIdentity.Call<int>("getBaseStationId", emptyObjectArr);
+        pair = new KeyValuePair<string, uint>(simpleName, (uint)baseStationId);
+      }
+      else
+      {
+        Debug.Log("Object is not an instance of a CellInfo class");
+      }
+
+      return pair;
+
+    }
+
+    public List<KeyValuePair<String, uint>> GetCellInfoList()
+    {
+      if (Application.platform != RuntimePlatform.Android)
+      {
+        Debug.Log("Not on android device.");
+        return null;
+      }
+
+      AndroidJavaObject telManager = GetTelephonyManager();
+      if (telManager == null)
+      {
+        Debug.Log("Can't get telephony manager!");
+        return null;
+      }
+
+      if (!Permission.HasUserAuthorizedPermission(Permission.FineLocation))
```

Comment:
This seems to be READ_PHONE_STATE?

Reply:
https://developer.android.com/reference/android/telephony/TelephonyManager.html#getAllCellInfo(). This only says Fine Location is required

Reply:
Great! Thanks for verifying.